test/QuantUtilsTest.cc

```diff
@@ -181,3 +181,32 @@ TEST_P(QuantizeGroupwiseTest, quantizeTest) {
   EXPECT_TRUE(isNear(dstint8, dstint8_ref));
   EXPECT_TRUE(isNear(dstint32, dstint32_ref));
 }
+
+TEST(QuantizeTest, cornerCases) {
+  TensorQuantizationParams qparams;
+  qparams.scale = 1.19209e-07;
+  qparams.zero_point = 0;
+  qparams.precision = 8;
+  std::vector<float> src1 = {3.40282e+38, -2.16845e+38};
+
+  std::vector<int8_t> dst_int8(src1.size());
+  Quantize<int8_t>(src1.data(), dst_int8.data(), dst_int8.size(), qparams);
+  EXPECT_EQ(dst_int8[0], 127);
+  EXPECT_EQ(dst_int8[1], -128);
+
+  // Tests vectorized and remainder paths
+  std::vector<float> src2 = {3.40282e+38,
```

Comment:
These are infinities? If the test is for the corner cases, it should cover other tests cases as well (s.a. extremely small, extremely small negative, etc.)

Reply:
I have postponed this diff for now. It needs more discussion on numeric mismatch between scalar and vector code.